components/map/basic/src/index.js

```diff
@@ -0,0 +1,257 @@
+      },
+      {
+        name: 'leaflet_map_poimouseout',
+        handleFunction: () => this.props.onPoiMouseOut()
+      },
+      {
+        name: 'leaflet_map_poimouseover',
+        handleFunction: (evt) => this.props.onPoiMouseOver(evt.detail)
+      },
+      {
+        name: 'leaflet_map_layer_satellite',
+        handleFunction: (evt) => this.props.onSatelliteView(evt.detail)
+      }
+    ]
+  }
+
+  getMapConfig () {
+    return {
+      id: this.props.id,
+      heatMapUrl: this.props.heatMapUrl,
+      latitude: this.props.center[0],
+      longitude: this.props.center[1],
+      literals: this.props.literals,
+      maxZoom: this.props.maxZoom,
+      minZoom: this.props.minZoom,
+      polygons: this.props.polygons,
+      showHeatmap: this.props.showHeatmap,
+      showSatelliteView: this.props.showSatelliteView,
+      mapViewModes: this.props.mapViewModes,
+      selectedMapViewMode: this.props.selectedMapViewMode,
+      zoomControl: this.props.zoomable,
+      zoomControlPosition: this.props.zoomControlPosition,
+      icons: this.props.icons,
+      enableViewMenu: this.props.enableViewMenu,
+      dragging: this.props.isInteractable,
+      zoom: this.props.zoom,
+      appId: this.props.appId,
+      appCode: this.props.appCode
+    }
+  }
+
+  subscribeToMapEvents () {
+    this.mapEventList.forEach(mapEvent => this.mapDOMInstance.addEventListener(mapEvent.name, mapEvent.handleFunction))
+  }
+
+  unsubscribeFromMapEvents () {
+    this.mapEventList.forEach(mapEvent => this.mapDOMInstance.removeEventListener(mapEvent.name, mapEvent.handleFunction))
+  }
+
+  checkWhichViewShouldBeDisplayed (showSatelliteView) {
+    if (showSatelliteView && !this.isSatelliteView) {
+      this.isSatelliteView = true
+      this.mapInstance.setView(mapViewModes.SATELLITE)
+    } else if (!showSatelliteView && this.isSatelliteView) {
+      this.isSatelliteView = false
+      this.mapInstance.setView(mapViewModes.NORMAL)
+    }
+  }
+
+  checkIfHeatMapShouldBeDisplayed (showHeatmap, url) {
+    if (showHeatmap && !this.isHeatmapVisible) {
+      this.isHeatmapVisible = true
+      this.mapInstance.showHeatMap(url)
+    } else if (!showHeatmap && this.isHeatmapVisible) {
+      this.isHeatmapVisible = false
+      this.mapInstance.removeHeatMapLayer()
+    }
+  }
+
+  componentWillUnmount () {
+    this.unsubscribeFromMapEvents()
+  }
+
+  shouldComponentUpdate () {
+    // The component itself have no changes. All changes are managed through leaflet maps api.
+    return false
+  }
+
+  componentWillReceiveProps (nextProps) {
+    const {heatMapUrl, pois, showHeatmap, showSatelliteView} = nextProps
+    pois.length && this.mapInstance.displayPois(pois)
+    this.checkIfHeatMapShouldBeDisplayed(showHeatmap, heatMapUrl)
+    this.checkWhichViewShouldBeDisplayed(showSatelliteView)
+  }
+
+  componentDidMount () {
+    this.subscribeToMapEvents()
+    this.mapInstance = new LeafletMap(this.getMapConfig())
```

Comment:
At this point, I think that you have the `this.mapDOMInstance` with the DOM element. It would be great to use this instead getting manually the element later in https://github.com/SUI-Components/sui-components/pull/223/files#diff-f4a8e7a6d35f7bb6b104560342f8ef5eR14

Also, you could pass down to the marker-manager.

Reply:
Fully agree with that. Next time I must check all the code on finish to re-refactor it.

Changed!